exchanges/mexc/mexc_wrapper.go

```diff
@@ -0,0 +1,1627 @@
+				AccountInfo:       true,
+				SubmitOrder:       true,
+			},
+			WebsocketCapabilities: protocol.Features{
+				TickerFetching:    true,
+				OrderbookFetching: true,
+				KlineFetching:     true,
+				AccountInfo:       true,
+				SubmitOrder:       true,
+			},
+			WithdrawPermissions: exchange.AutoWithdrawCrypto |
+				exchange.AutoWithdrawFiat,
+		},
+		Enabled: exchange.FeaturesEnabled{
+			AutoPairUpdates: false,
+			Kline: kline.ExchangeCapabilitiesEnabled{
+				Intervals: kline.DeployExchangeIntervals(
+					kline.IntervalCapacity{Interval: kline.OneMin},
+					kline.IntervalCapacity{Interval: kline.FiveMin},
+					kline.IntervalCapacity{Interval: kline.FifteenMin},
+					kline.IntervalCapacity{Interval: kline.ThirtyMin},
+					kline.IntervalCapacity{Interval: kline.OneHour},
+					kline.IntervalCapacity{Interval: kline.FourHour},
+					kline.IntervalCapacity{Interval: kline.OneDay},
+					kline.IntervalCapacity{Interval: kline.OneWeek},
+					kline.IntervalCapacity{Interval: kline.OneMonth},
+				),
+				GlobalResultLimit: 1000,
+			},
+		},
+		Subscriptions: defaultSubscriptions.Clone(),
+	}
+	var err error
+	e.Requester, err = request.New(e.Name,
+		common.NewHTTPClientWithTimeout(exchange.DefaultHTTPTimeout),
+		request.WithLimiter(GetRateLimit()),
+	)
+	if err != nil {
+		log.Errorln(log.ExchangeSys, err)
+	}
+
+	e.API.Endpoints = e.NewEndpoints()
+	if err := e.API.Endpoints.SetDefaultEndpoints(map[exchange.URL]string{
+		exchange.RestSpot:              spotAPIURL,
+		exchange.RestFutures:           contractAPIURL,
+		exchange.WebsocketSpot:         spotWebsocketURL,
+		exchange.WebsocketUSDTMargined: futuresWebsocketURL,
+	}); err != nil {
+		log.Errorln(log.ExchangeSys, err)
+	}
+	e.Websocket = websocket.NewManager()
+	e.WebsocketResponseMaxLimit = exchange.DefaultWebsocketResponseMaxLimit
+	e.WebsocketResponseCheckTimeout = exchange.DefaultWebsocketResponseCheckTimeout
+	e.WebsocketOrderbookBufferLimit = exchange.DefaultWebsocketOrderbookBufferLimit
+}
+
+// Setup takes in the supplied exchange configuration details and sets params
+func (e *Exchange) Setup(exch *config.Exchange) error {
+	if err := exch.Validate(); err != nil {
+		return err
+	}
+	if !exch.Enabled {
+		e.SetEnabled(false)
+		return nil
+	}
+	if err := e.SetupDefaults(exch); err != nil {
+		return err
+	}
+	spotWSURL, err := e.API.Endpoints.GetURL(exchange.WebsocketSpot)
+	if err != nil {
+		return err
+	}
+	futuresWSURL, err := e.API.Endpoints.GetURL(exchange.WebsocketUSDTMargined)
+	if err != nil {
+		return err
+	}
+	if err := e.Websocket.Setup(&websocket.ManagerSetup{
+		ExchangeConfig: exch,
+		Features:       &e.Features.Supports.WebsocketCapabilities,
+		DefaultURL:     spotWebsocketURL,
+		RunningURL:     spotWebsocketURL,
+		OrderbookBufferConfig: buffer.Config{
+			SortBuffer:            true,
+			SortBufferByUpdateIDs: true,
+		},
+		TradeFeed:                    e.Features.Enabled.TradeFeed,
+		UseMultiConnectionManagement: true,
```

Comment:
Please add `MaxWebsocketSubscriptionsPerConnection`: `One ws connection supports a maximum of 30 subscriptions.`

See: https://www.mexc.com/api-docs/spot-v3/websocket-market-streams